scan-engine/src/androidTest/java/de/tillhub/scanengine/barcode/SunmiBarcodeScannerTest.kt

```diff
@@ -0,0 +1,59 @@
+package de.tillhub.scanengine.barcode
+
+
+import android.content.Context
+import android.content.Intent
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import de.tillhub.scanengine.data.ScanEvent
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNull
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.lang.Thread.sleep
+
+@ExperimentalCoroutinesApi
+@RunWith(AndroidJUnit4::class)
+class SunmiBarcodeScannerTest {
+
+    private lateinit var context: Context
+    private lateinit var mutableScanEvents: MutableSharedFlow<ScanEvent>
+
+    @Before
+    fun setUp() {
+        context = ApplicationProvider.getApplicationContext()
+        mutableScanEvents = MutableSharedFlow(replay = 1)
+        SunmiBarcodeScanner(context, mutableScanEvents)
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun testBroadcastReceiverRegistered() {
+        val intent = Intent(SunmiBarcodeScanner.ACTION_DATA_CODE_RECEIVED).apply {
```

Comment:
In tests use hard coded string here "com.sunmi.scanner.ACTION_DATA_CODE_RECEIVED" we don't want to use variable from actual code.